src/connection/WalletSDKConnection.ts

```diff
@@ -93,7 +93,7 @@ export class WalletSDKConnection {
       ws.connectionState$
         .pipe(
           tap(state =>
-            this.eventListener?.onEvent(EVENTS.CONNECTED_STATE_CHANGE, {
+            this.diagnosticLogger?.log(EVENTS.CONNECTED_STATE_CHANGE, {
```

Comment:
nit: `diagnosticLogger.log()` seems a bit redundant. I think `diagnostic.log()` reads a bit better

Reply:
eth/tracers/api.go

Reply:
yeah I agree, struggled with the naming a bit here

Reply:
I kept the interface name as DiagnosticLogger but changed the property name in each of the classes that use it.  Do you think that makes sense or should I change it everywhere?